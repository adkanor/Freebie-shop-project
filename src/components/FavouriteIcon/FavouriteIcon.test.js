import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import FavoriteIcon from "./FavouriteIcon";
import "@testing-library/jest-dom/extend-expect";

const mockStore = configureStore([]);
describe("FavoriteIcon Component", () => {
    it("renders the FavoriteIcon component", () => {
        const initialState = {
            favoritesReducer: {
                favorites: [],
            },
        };
        const store = mockStore(initialState);

        const component = renderer.create(
            <Provider store={store}>
                <FavoriteIcon thisCard={{ id: 1 }} />
            </Provider>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
