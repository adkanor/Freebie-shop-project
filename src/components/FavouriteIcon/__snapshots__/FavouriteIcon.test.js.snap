// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteIcon Component renders the FavoriteIcon component 1`] = `null`;
